client/src/components/ProtectedRoute/ProtectedRoute.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect } from 'react';
+import { useAuth } from '../../context/useAuthContext';
+import { useSocket } from '../../context/useSocketContext';
+import CircularProgress from '@mui/material/CircularProgress';
+import { Route, Redirect, RouteProps, RouteComponentProps } from 'react-router-dom';
+
+const ProtectedRoute: React.FC<RouteProps> = ({ component, render, ...routeProps }) => {
```

Comment:
```suggestion
const ProtectedRoute = ({ component: Component, ...rest }: RouteProps) : JSX.Element  => {
```

Short list of React.FC cons:

- Provides an implicit definition of children, even if your component doesn't need to have children. That might cause an error.
- Doesn't support generics.
- Doesn't work correctly with defaultProps.

Additional Reference: https://github.com/facebook/create-react-app/pull/8177